Sources/BraintreePaymentFlow/BTPaymentFlowClient.swift

```diff
@@ -20,23 +20,36 @@ import BraintreeCore
     private var paymentFlowName: String {
         return paymentFlowRequestDelegate?.paymentFlowName() ?? "local-payments"
     }
-    
+    private var isAuthenticationSessionStarted: Bool = false
+    private var returnedToAppAfterPermissionAlert: Bool = false
+
     // MARK: - Public Methods
     
     /// Initialize a new BTPaymentFlowClient instance.
     /// - Parameter apiClient: An API client
     @objc(initWithAPIClient:)
     public init(apiClient: BTAPIClient) {
         self._apiClient = apiClient
+        super.init()
+        NotificationCenter.default.addObserver(
+                    self,
+                    selector: #selector(applicationDidBecomeActive),
+                    name: UIApplication.didBecomeActiveNotification,
+                    object: nil
+                )
     }
     
+    @objc func applicationDidBecomeActive(notification: Notification) {
+           returnedToAppAfterPermissionAlert = isAuthenticationSessionStarted
+       }
+    
     /// Starts a payment flow using a BTPaymentFlowRequest (usually subclassed for specific payment methods).
     /// - Parameters:
     ///   - request: A BTPaymentFlowRequest request.
     ///   - completionBlock: This completion will be invoked exactly once when the payment flow is complete or an error occurs.
     public func startPaymentFlow(_ request: BTPaymentFlowRequest & BTPaymentFlowRequestDelegate, completion: @escaping (BTPaymentFlowResult?, Error?) -> Void) {
         setupPaymentFlow(request, completion: completion)
-        _apiClient.sendAnalyticsEvent("ios.\(paymentFlowName).start-payment.selected")
+        sendAnalyticsEvent(BTPaymentFlowAnalytics.paymentStarted)
```

Comment:
@scannillo I realize that I am sending "three-d-secure:start-payment:started" here for 3Ds flow which deviates from "3ds:verify:started" that you specified in the list of events for 3DS.

Reply:
I guess I can have two sets of analytic event names in BTFlowAnalytics and use a helper function to send the appropriate name depending not he flowName.

Reply:
It's fine if the 3DS events are wonky until you get to it in your 3DS PR!